Update run script SHA check

adam-h opened the pull request:
Newer redis (redis-rb@5, redis@6.2) don't return the sha from evalsha

adam-h committed:
Update run script SHA check

Newer redis (redis-rb@5, redis@6.2) don't return the sha from evalsha

Using new pipeline syntax which requires calling redis methods on the block argument

Co-authored-by: Brad Lindsay <sluggy.fan@gmail.com>

bfad commented on lib/redis_scripts.rb:
```diff
@@ -57,8 +57,9 @@ def run(name, *args)
       sha, value = redis.pipelined do
-        redis.script 'load', script.content
-        redis.evalsha(script.sha, *args)
+        sha = redis.script 'load', script.content
+        result = redis.evalsha(script.sha, *args)
+        return sha, result
```
Won't `return` in this context exit you from the `run` method itself? It might not as `pipelined` may take the block and convert it into a lambda before running, but that complexity is hidden, so I would do the following instead:

```suggestion
        [redis.script('load', script.content), redis.evalsha(script.sha, *args)]
```

bfad reviewed (changes requested):
I have one concern below which you may have tested for, but it looks suspicious for how blocks are normally called in Ruby.

bfad commented:
Actually, I have one additional concern. It looks like this doesn't work with both older and newer versions of our redis gem. If the older version returns the SHA first, then the SHA will be in both `sha` and `result` for older versions of the gem. We need code that will work for both versions while we transition.

adam-h committed:
Switch to redis-rb@5

`redis.evalsha` just returns the script result now, not the SHA. So we
instead check the SHA from the `redis.load` command.

Note this no longer supports redis-rb@4

adam-h commented:
@bfad Thanks for catching that. It did work, but you're right it does seem to rely on some undocumented magic!

Switched this branch out to now explicitly only support `redis-rb`@5 so we can update this gem with redis in our rails app.

bfad reviewed (approved):
LGTM
